### PR TITLE
[FEATURE] Add command to generate self-signed certs in authelia binary.

### LINKED ISSUE #590: Provide self-signed certificate to quickly deploy a test instance of Authelia
Hi,
I'm attempting to setup Authelia on my home server and I'm not being correctly redirected to my service after successfully logging in.
I've traced the `Required level for the URL http://droppy.myserver.lan/ is 1` message back to the following lines in the code: https://github.com/authelia/authelia/blob/e6ddedf23de4aa6bb26dc9cf30ec86d6ab6b16b5/internal/handlers/handler_firstfactor.go#L125-L135

Which I then followed to https://github.com/authelia/authelia/blob/3b2d733367c88621e4178301f2bcb4bc03613eee/internal/handlers/safe_redirection.go#L8-L17

It doesn't seem possible to redirect back to non-http urls. Is there a configuration option I'm missing, or am I S.O.L?

### PR DESCRIPTION
**Pull request by clems4ever**
Fixes #590.

### CODE REVIEW

File: docs/deployment/deployment-lite.md [hunk 1]
```diff
@@ -19,6 +19,18 @@ only two: a reverse proxy such as Nginx, Traefik or HAProxy and Authelia.
 Documentation for deploying a reverse proxy collaborating with Authelia is available
 [here](./supported-proxies/index.md).
 
+Please note that Authelia only works for websites served over HTTPS because the session cookie
+can only be transmitted over secure connections. Therefore, would you need to generate a
```
Comment (nightah) [inline docs/deployment/deployment-lite.md]:
```suggestion
can only be transmitted over secure connections. Therefore, you need to generate a
```

File: docs/deployment/deployment-lite.md [hunk 2]
```diff
@@ -19,6 +19,18 @@ only two: a reverse proxy such as Nginx, Traefik or HAProxy and Authelia.
...
+self-signed certificate for your setup, you can use the dedicated helper function provided
+by authelia binary.
```
Comment (nightah) [inline docs/deployment/deployment-lite.md]:
```suggestion
by the authelia binary.
```

File: docs/deployment/deployment-lite.md [hunk 3]
```diff
@@ -19,6 +19,18 @@ only two: a reverse proxy such as Nginx, Traefik or HAProxy and Authelia.
...
+
+    # Generate a certificate covering "example.com" for one year and stores it in /tmp/certs/ directory.
```
Comment (nightah) [inline docs/deployment/deployment-lite.md]:
```suggestion
    # Generate a certificate covering "example.com" for one year in the /tmp/certs/ directory.
```

File: docs/home/architecture.md [hunk 4]
```diff
@@ -46,3 +46,14 @@ Here is a description of the complete workflow:
...
+## HTTP/HTTPS
+
+Authelia only works for websites served over HTTPS because the session cookie can only be
+transmitted over secure connections. Please note that it has been decided that we won't
+support websites served over HTTP in order to avoid any risk due to misconfiguration.
+(see [#590](https://github.com/authelia/authelia/issues/590)).
+
+If a self-signed certificate is required, the following command can be used to generate one:
+
+    # Generate a certificate covering "example.com" for one year and stores it in /tmp/certs/ directory.
```
Comment (nightah) [inline docs/home/architecture.md]:
```suggestion
    # Generate a certificate covering "example.com" for one year in the /tmp/certs/ directory.
```

File: internal/commands/certificates.go [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+func init() {
+	CertificatesGenerateCmd.PersistentFlags().StringVar(&host, "host", "", "Comma-separated hostnames and IPs to generate a certificate for")
+	err := CertificatesGenerateCmd.MarkPersistentFlagRequired("host")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	CertificatesGenerateCmd.PersistentFlags().StringVar(&validFrom, "start-date", "", "Creation date formatted as Jan 1 15:04:05 2011")
+	CertificatesGenerateCmd.PersistentFlags().DurationVar(&validFor, "duration", 365*24*time.Hour, "Duration that certificate is valid for")
+	CertificatesGenerateCmd.PersistentFlags().BoolVar(&isCA, "ca", false, "whether this cert should be its own Certificate Authority")
```
Comment (nightah) [inline internal/commands/certificates.go]:
```suggestion
	CertificatesGenerateCmd.PersistentFlags().BoolVar(&isCA, "ca", false, "Whether this cert should be its own Certificate Authority")
```

File: internal/commands/certificates.go [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+	CertificatesGenerateCmd.PersistentFlags().IntVar(&rsaBits, "rsa-bits", 2048, "Size of RSA key to generate. Ignored if --ecdsa-curve is set")
+	CertificatesGenerateCmd.PersistentFlags().StringVar(&ecdsaCurve, "ecdsa-curve", "", "ECDSA curve to use to generate a key. Valid values are P224, P256 (recommended), P384, P521")
+	CertificatesGenerateCmd.PersistentFlags().BoolVar(&ed25519Key, "ed25519", false, "Generate an Ed25519 key")
+
```
Comment (nightah) [inline internal/commands/certificates.go]:
```suggestion
```

File: docs/deployment/deployment-lite.md (hunk 1 above)
Comment (clems4ever) [reply docs/deployment/deployment-lite.md]: I wanted to let the user know that she "can", not that she "should". What do you think about `Therefore, if you need to generate a...`.

Comment (nightah) [reply docs/deployment/deployment-lite.md]: Yep just `Therefore, if you need to`.

Comment (clems4ever) [reply docs/deployment/deployment-lite.md]: Good.

Comment (clems4ever) [inline docs/deployment/deployment-lite.md]:
```suggestion
can only be transmitted over secure connections. Therefore, if you need to generate a
```

**Pull request merged**